server/channels/db/migrations/mysql/000121_drop_threads_teamid.up.sql

```diff
@@ -0,0 +1,14 @@
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Threads'
+        AND table_schema = DATABASE()
+        AND column_name = 'TeamId'
+    ) > 0,
+    'ALTER TABLE Threads DROP COLUMN TeamId;',
+    'SELECT 1'
+));
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
```

Comment:
I understand this will make the tables consistent, but this also means downtime during the upgrade. Would we consider leaving the column and considering it deprecated instead? Or if it's causing an issue, can we relax any non-null requirements so that the current application (which doesn't write to it) won't trigger these? Anything to avoid the table rewrite.

Reply:
@lieut-data this issue was discovered during a customer migration. We can't specifically ignore a column easily. What we can do is basically we can try relaxing this issue by a) entirely ignoring the table during the migration and creating views ([suggestion](https://stackoverflow.com/a/45690787) from the original author) b) dropping the column just before the migration with a `BEFORE LOAD` clause but that would be against of being migration to be safe. c) create the same column in postgres which would be more problematic IMHO.

Neither of these seemed efficient to me but I'm open for alternatives and further discussions.

Reply:
I agree with Jesse. We cannot make blocking migrations any more unfortunately. Both a or b sound like reasonable alternatives to me.

Reply:
Thanks for the feedback @lieut-data @agnivade what if we used `ALGORITHM=INSTANT`? We already have mysql-8 requirement anyway.